src/_macosx.m

```diff
@@ -2066,29 +2066,31 @@ - (BOOL)acceptsFirstResponder
     return YES;
 }
 
-/* This is all wrong. Address of pointer is being passed instead of pointer, keynames don't
-   match up with what the front-end and does the front-end even handle modifier keys by themselves?
-
+// flagsChanged gets called on single modifier keypresses
+// The modifier + second key gets handled in convertKeyEvent in KeyUp/KeyDown
 - (void)flagsChanged:(NSEvent*)event
 {
-    const char *s = NULL;
-    if (([event modifierFlags] & NSControlKeyMask) == NSControlKeyMask)
+    const char* s = NULL;
+    if (([event modifierFlags] & NSEventModifierFlagControl) == NSEventModifierFlagControl)
```

Comment:
Also, you can just check the boolean state of the & (without the equality check); this is shorter and consistent with the implementation in convertKeyEvent.

Reply:
:+1:, blindly copying examples ;) Not sure why they do it that way in many online examples then. I agree that is simpler and makes sense.